packages/exhibitor/extra/dcos_zk_backup.py

```diff
@@ -0,0 +1,350 @@
+    # while running, no data is lost inplace.
+    # However, this has a downside that we may use a lot of disk space -
+    # we have seen ZooKeeper files up to 20 GB in size.
+    _copy_dir_and_preserve_ownership(
+        src=zookeeper_dir,
+        dst=tmp_zookeeper_dir,
+        verbose=verbose,
+    )
+    shutil.rmtree(path=str(zookeeper_dir), ignore_errors=True)
+
+    print('Restoring {zookeeper_dir} from backup {backup}'.format(
+        zookeeper_dir=zookeeper_dir,
+        backup=backup,
+    ))
+    with tarfile.open(name=backup, mode='r:gz') as tar:
+        tar.extractall(path=str(EXHIBITOR_DIR))
+        if verbose:
+            tar.list()
+
+    print('Deleting temporary files in {tmp_zookeeper_dir}'.format(
+        tmp_zookeeper_dir=tmp_zookeeper_dir,
+    ))
+    shutil.rmtree(path=str(tmp_zookeeper_dir), ignore_errors=True)
+
+    print('Local ZooKeeper instance restored successfully')
+
+
+def _non_existing_file_path_existing_parent_dir(value: str) -> Path:
+    """
+    Validate that the value is a path to a file which does not exist
+    but the parent directory tree exists.
+    """
+    print(type(value))
+    path = Path(value)
+    if path.exists():
+        raise ArgumentTypeError('{} already exists'.format(path))
+    if not Path(path.parent).exists():
+        raise ArgumentTypeError(
+            '{} parent directory does not exist'.format(path),
+        )
+    return path.absolute()
+
+
+def _existing_file_path(value: str) -> Path:
+    """
+    Validate that the value is a file which does exist on the file system.
+    """
+    path = Path(value)
+    if not path.exists():
+        raise ArgumentTypeError('{} does not exist'.format(path))
+    if not path.is_file():
+        raise ArgumentTypeError('{} is not a file'.format(path))
+    return path.absolute()
+
+
+def _existing_dir_path(value: str) -> Path:
+    """
+    Validate that the value is a directory which does exist on the file system.
+    """
+    path = Path(value)
+    if not path.exists():
+        raise ArgumentTypeError('{} does not exist'.format(path))
+    if not path.is_dir():
+        raise ArgumentTypeError('{} is not a directory'.format(path))
+    return path.absolute()
+
+
+class DCOSZooKeeperCli:
+    """
+    Minimal CLI to backup/restore the local DC/OS ZooKeeper instance.
+    """
+
+    def __init__(self) -> None:
+        """
+        Present CLI command choices.
+        """
+        parser = ArgumentParser(
+            description=(
+                'Command line utility to more safely interact with the local '
+                'ZooKeeper instance on DC/OS master nodes'
+            )
+        )
+        parser.add_argument(
+            'command',
+            type=str,
+            choices=[
+                'backup',
+                'restore',
+            ],
+            help='CLI commands available',
+        )
+        args = parser.parse_args(sys.argv[1:2])
+        getattr(self, args.command)()
```

Comment:
@timaa2k We discussed using `click` but concluded that you would rather not add a dependency for this script. However, I see we already have `click` as an [indirect dependency](https://github.com/dcos/dcos/blob/1c8088873cb8cbe5abb514deea35375f2a08b313/packages/flask/buildinfo.json#L9) and we import it directly:

https://github.com/dcos/dcos/blob/c71cf838783ee7f69dfb8d51cda01f5f03c6b025/packages/dcos-integration-test/extra/get_test_group.py#L16

Can we use it here?

Reply:
I should merge as is once the tests pass! Let's do it once we add another command.